lib/matplotlib/tests/test_gridspec.py

```diff
@@ -1,8 +1,17 @@
 import matplotlib.gridspec as gridspec
-from nose.tools import assert_equal
+from nose.tools import assert_raises, assert_equal
 
 
 def test_equal():
     gs = gridspec.GridSpec(2, 1)
     assert_equal(gs[0, 0], gs[0, 0])
     assert_equal(gs[:, 0], gs[:, 0])
+
+
+def test_ratios():
+    """
+    Addresses issue #5835.
+    See at https://github.com/matplotlib/matplotlib/issues/5835.
+    """
+    assert_raises(ValueError, gridspec.GridSpec,
```

Comment:
Can you add a second one of these which fails via the other path (if I am reading this right this only really tests the width based failure).